### PR TITLE
Service Accounts - Fix delete token status code

### PR DESCRIPTION
**Pull request by ywangd**
The delete token response now returns status code 404 instead of 200
when the token does not exist.

### CODE REVIEW

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -158,7 +158,8 @@ tasks.named("yamlRestCompatTest").configure {
     'vectors/40_sparse_vector_special_cases/Query vector has different dimensions from documents\' vectors',
     'vectors/40_sparse_vector_special_cases/Sparse vectors should error with dense vector functions',
     'vectors/40_sparse_vector_special_cases/Vectors of different dimensions and data types',
-    'vectors/50_vector_stats/Usage stats on vector fields'
+    'vectors/50_vector_stats/Usage stats on vector fields',
+    'service_accounts/10_basic/*'
```
Comment (ywangd) [inline x-pack/plugin/build.gradle:162]: Had to disable restCompt for this temporarily. Will enable it again once backported.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestDeleteServiceAccountTokenAction.java [hunk 2]
```diff
@@ -47,9 +52,15 @@ protected RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClien
...
-            new RestToXContentListener<>(channel));
+        return channel -> client.execute(DeleteServiceAccountTokenAction.INSTANCE, deleteServiceAccountTokenRequest,
+            new RestBuilderListener<>(channel) {
+                @Override
+                public RestResponse buildResponse(DeleteServiceAccountTokenResponse response, XContentBuilder builder) throws Exception {
+                    return new BytesRestResponse(response.found() ? RestStatus.OK : RestStatus.NOT_FOUND,
+                        builder.startObject()
+                            .field("found", response.found())
+                            .endObject());
+                }
```
Comment (tvernum) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestDeleteServiceAccountTokenAction.java:63]: This is fine, but you can also do
```
new RestToXContentListener<>(channel) {
  protected RestStatus getStatus(Response response) {
      return response.found() ? RestStatus.OK : RestStatus.NOT_FOUND;
  }
}
```

**Review by tvernum — APPROVED** [review]

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestDeleteServiceAccountTokenAction.java (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestDeleteServiceAccountTokenAction.java]: Updated, thanks!

**Comment by ywangd** [comment]
Replaced `>bug` with `>non-issue` since it is not released.

**Pull request merged**